pyhunter/pyhunter.py

```diff
@@ -14,10 +14,10 @@ def _query_hunter(self, endpoint, params, request_type='get',
 
         request_kwargs = dict(params=params)
         if payload:
-            request_kwargs.setdefault(json=payload)
+            request_kwargs.setdefault('json', payload)
 
         if headers:
-            request_kwargs.setdefault(headers=headers)
+            request_kwargs.setdefault('headers', headers)
```

Comment:
I'm not completely sure about this change.
 `setdefault(json=payload)` throws this error for me: `TypeError: dict.setdefault() takes no keyword arguments`

Reply:
This part of the code is not from me, I need to get back at it before to answer you with some stupid fix haha

Reply:
I just checked what happened there.

The bug was introduced in PR https://github.com/VonStruddle/PyHunter/pull/1#pullrequestreview-570565505
It was later reported in issue https://github.com/VonStruddle/PyHunter/issues/53

@dimitern wanted to factor out `request_kwargs` in order to add `headers`. But they got the `dict.setdefault` method wrong.

My fix should be ok, but after seeing how it's being used (if it's None it's not added, and then the dict is sent to request as its kwargs, but request adds None by default to all arguments), I realized that there's a much simpler way of doing this: `request_kwargs = dict(params=params, json=payload, headers=headers)`

Reply:
@VonStruddle you can merge now if you agree 👍